### ISSUE TITLE
Integrate SpectralCoord with Spectrum1D

### ISSUE THREAD

**Issue by rosteen** [astropy/specutils#599]
The `Spectrum1D` class will be updated to accept `SpectralCoord` objects as `spectral_axis` input, and internally convert `spectral_axis` inputs provided as `Quantity`s to `SpectralCoord` objects.

**Comment by rosteen** [comment]
Depends on [https://github.com/astropy/specutils/pull/524](https://github.com/astropy/specutils/pull/524)

**Comment by rosteen** [comment]
@eteq @nmearl A question about the intent here. Is the desired behavior:

(a) That one can pass a SpectralCoord to build a Spectrum1D object (possibly packaging e.g. radial_velocity inside the SpectralCoord instead of as an argument for Spectrum1D), but that the resulting Spectrum1D object remain unchanged, i.e if one were to look at spectrum1d_object.spectral_axis it would be a Quantity,

or 

(b) That however the input information is passed in, spectrum1d_object.spectral_axis ends up being a SpectralCoord object with any relevant information attached (e.g. radial_velocity if it was passed as an argument to Spectrum1D)?

**Comment by rosteen** [comment]
Oh, or of course (c) the resulting Spectrum1D object ends up with whichever spectral_axis format was passed in, a simple option that I forgot about while perhaps thinking too hard about this.

**Comment by nmearl** [comment]
The second. The associated radial velocity/redshift arguments to `Spectrum1D` will remain, but the initializer will instantiate a `SpectralCoord` passing in that information. The `SpectralCoord` object will exist as some private variable (`self._spectral_coord`) that should be returned from a property of the same name:

```
@property
def spectral_coord(self):
    return self._spectral_coord
```

The real tough part of this is that we have not discussed how the spectral coordinate will handle wcs information. Currently, no spectral axis information is actually maintained on the `Spectrum1D` object, it's lazily evaluated from the wcs's `pixel_to_world` method.

@eteq, @astrofrog, do we wanted to switch to storing wcs objects on the `SpectralCoord`?

**Comment by rosteen** [comment]
Got it, thanks. This was indeed motivated by my realizing that currently a spectral_axis is just used to create the WCS, ditched, and re-instantiated from the WCS if needed again, which I'd need to bypass to keep the spectral coord object around.

**Comment by rosteen** [comment]
Should the updated version of the spectral_axis property live in Spectrum1D to override that in OneDSpectrumMixin, or do we want to update it at the higher level there? It doesn't look like anything other than Spectrum1D inherits from the mixin so I imagine I can change it there, but I figured I'd check.

**Comment by nmearl** [comment]
Yes, it should be changed in the mixin, not the `Spectrum1D`.

**Comment by rosteen** [comment]
Resolved by [PR #610 ](https://github.com/astropy/specutils/pull/610)
